src/main/scala/com/fulcrumgenomics/bam/ComputeRnaSeqInsertSize.scala

```diff
@@ -0,0 +1,330 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import java.io.{Closeable, File, InputStream}
+
+import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.bam.GeneAnnotations._
+import com.fulcrumgenomics.cmdline.{ClpGroups, FgBioTool}
+import com.fulcrumgenomics.util._
+import dagr.commons.CommonsDef.{PathToBam, PathToFastq}
+import dagr.commons.util.LazyLogging
+import dagr.sopt.{arg, clp}
+import htsjdk.samtools.SamPairUtil.PairOrientation
+import htsjdk.samtools._
+import htsjdk.samtools.util.{CoordMath, Interval, OverlapDetector}
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions.iterableAsScalaIterable
+import scala.io.Source
+
+@clp(description =
+  """Computes the insert size for RNA-Seq experiments.
+    |
+    |Computes the insert size be counting the # of bases sequenced in transcript space.  Skips reads that overlap
+    |multiple genes, and reads where the insert size would disagree across transcripts from the same gene.
+    |
+    |Also skips unpaired, failed QC, secondary, supplementary, pairs without both ends mapped, duplicates, and pairs whose
+    |reads map to different chromosomes.
+    |
+    |This tool requires each mapped pair to have the mate cigar (MC) tag and for read pairs not to overlap.  Use Picard's
+    |RevertOriginalBaseQualitiesAndAddMateCigar to add the former and Fgbio's ClipOverlappingReads to ensure the latter.
+  """,
+  group = ClpGroups.SamOrBam)
+class ComputeRnaSeqInsertSize
+(@arg(flag="i", doc="Input BAM file.") val input: PathToBam,
+ @arg(flag="r", doc="Input gene annotations in refFlat form.  Format described here: http://genome.ucsc.edu/goldenPath/gbdDescriptionsOld.html#RefFlat")
```

Comment:
I started going down the road of writing a GTF->refFlat parser, but it was quite tedious with all the book keeping.  It's not needed so I will defer, but if you have a case for needing it, I can share my unfinished code.